### ISSUE TITLE
sparkNetwork.doEvaluation() return recall, precision are nan

### ISSUE THREAD

**Issue by alg-jmx** [deeplearning4j/deeplearning4j#4471]
[this is my code link:](https://gist.github.com/alg-jmx/cbb9c2ad4a7cda9cd8066ffe1954dd58)
![image](https://user-images.githubusercontent.com/11658913/34567305-b48a9716-f19c-11e7-9007-56699829e48d.png)
the code runing on spark on yarn

**Comment by AlexDBlack** [comment]
Two other things that would really help me reproduce and fix this much quicker:
1. Provide output of Evaluation.toJson()
2. Provide serialized Evaluation object file, as per the code below.

```
        ObjectOutputStream oos = new ObjectOutputStream(new FileOutputStream(new File("eval.bin")));
        oos.writeObject(evaluation);
        oos.flush();
        oos.close();
```

**Comment by AlexDBlack** [comment]
This looks exactly like a known issue on 0.9.1 that was fixed in 1.0.0-alpha
https://deeplearning4j.org/releasenotes#zeronineone

If this still occurs with 1.0.0-alpha, we can re-open this issue.
